wordpress-mobile/WordPress-Android
# Default transition animations

Closes #4377. As part of the tablet refresh project, we decided to remove the slide-in/out animations and just use the default ones. It'll be much easier to handle when we have more than one fragment visible. Also, slide-in/out creates the illusion that the other page is living on the left or right, but multiple visible fragments break that illusion. The only slide-in/out animation left in the app is for the site-picker as per @nbradbury's suggestion. I agree with the sentiment that it's better to have a different animation for that one and we can change it to something else later on if we feel the need for it.

* I wasn't completely sure about the change for the `openNote` because we were only using the sliding animation in some cases. @nbradbury your commit says `Don't slide in note detail when opened from a push`, so I am guessing it's fine to disable it for all cases.
* Since we were using multiple fragments in People Management, we had a bunch of code just to have the sliding animation. I've 🔥 all of them.

To test:
* Just browse the app and check the animations. 
* I've already tested it with my Samsung S6 (6.0.1), Genymotion devices (4.4.4 & 6.0.0) and with an Android N simulator. I've tried to hit every page, but we have a lot of them, so hopefully I didn't miss any.
* It'd be good to pay a close attention to the People page when you select a person from the list. It looks like the animation there is too fast/non-existing. I was wondering if we should add anything specific there, but when I tried fade-in/out it was really bad.

/cc @hypest, @mattmiklic, @astralbodies 

Needs review: @nbradbury
